File: backend/src/main/java/dev/coldhands/pair/stairs/backend/domain/DeveloperInfo.java
package dev.coldhands.pair.stairs.backend.domain;

import java.util.Comparator;

// todo make kotlin
public record DeveloperInfo(long id, String displayName, boolean archived) implements Comparable<DeveloperInfo> {

    public static final Comparator<DeveloperInfo> COMPARATOR = Comparator.comparing(DeveloperInfo::displayName);

    @Override
    public int compareTo(DeveloperInfo o) {
        return COMPARATOR.compare(this,o);
    }
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/domain/developer/DeveloperInfo.kt
package dev.coldhands.pair.stairs.backend.domain.developer

import dev.coldhands.pair.stairs.backend.domain.DeveloperId

// todo should this be called a Dto?
data class DeveloperInfo(
    val id: DeveloperId,
    val displayName: String,
    val archived: Boolean,
): Comparable<DeveloperInfo> {
    override fun compareTo(other: DeveloperInfo): Int {
        return compareBy<DeveloperInfo> { it.displayName }.compare(this, other)
    }

}
